tests/peer/meshRoom.js

```diff
@@ -516,6 +516,79 @@ describe('MeshRoom', () => {
         }
       });
     });
+
+    it('should emit a brodcast evnet at intervals greater than limit even if send intervals are smaller than limit', async () => {
+      const data = 'foobar';
+
+      const sendIntervalMs = 10;
+      const broadcastInterval = 100;
+      const toleranceRange = 5;
+
+      let lastEventTime = 0;
+      meshRoom.on(MeshRoom.MESSAGE_EVENTS.broadcast.key, () => {
+        const now = Date.now();
+        const diff = now - lastEventTime;
+        if (diff < broadcastInterval - toleranceRange) {
+          assert.fail(`broadcast event is emitted too early: ${diff} ms`);
+        }
+        lastEventTime = now;
+      });
+
+      const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));
+      for (let i = 0; i < 10; ++i) {
+        meshRoom.send(data);
+        await sleep(sendIntervalMs);
+      }
+    });
+
+    it('should queue if message queue is not empty', async () => {
+      const data = 'foobar';
+
+      const sendIntervalMs = 110;
+      const broadcastInterval = 100;
+      const toleranceRange = 5;
+
+      let lastEventTime = 0;
+      meshRoom.on(MeshRoom.MESSAGE_EVENTS.broadcast.key, () => {
+        const now = Date.now();
+        const diff = now - lastEventTime;
+        if (diff < broadcastInterval - toleranceRange) {
+          assert.fail(`broadcast event is emitted too early: ${diff} ms`);
+        }
+        lastEventTime = now;
+      });
+
+      const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));
+      for (let i = 0; i < 5; ++i) {
+        meshRoom.send(data);
+      }
+      for (let i = 0; i < 5; ++i) {
+        meshRoom.send(data);
+        await sleep(sendIntervalMs);
+      }
+    });
+
+    it('should emit a broadcaset event without delay if send intervals are greter than limit', async () => {
+      const data = 'foobar';
+
+      const sendIntervalMs = 150;
+      const broadcastMaxDelayMs = sendIntervalMs * 10 + 100;
```

Comment:
Can you tell me why you set broadcastMaxDelayMs to this value?

Reply:
We send a message 10 times in this test. Therefore, we need to wait at least `sendIntervalMs * 10`.  Moreover, we expect that messages sent in this test are not delayed.  This is why we add 100ms to expected max delay.